router/src/dml_handlers/rpc_write.rs

```diff
@@ -14,12 +14,17 @@ use observability_deps::tracing::*;
 use sharder::RoundRobin;
 use std::{fmt::Debug, time::Duration};
 use thiserror::Error;
-use tonic::transport::Channel;
 use trace::ctx::SpanContext;
 
 /// Create a client to the ingester's write service.
-pub fn write_service_client(ingester_addr: &str) -> WriteServiceClient<Channel> {
-    WriteServiceClient::new(Channel::builder(ingester_addr.parse().unwrap()).connect_lazy())
+pub async fn write_service_client(
+    ingester_addr: &str,
+) -> WriteServiceClient<client_util::connection::GrpcConnection> {
+    let connection = client_util::connection::Builder::default()
+        .build(format!("http://{}", ingester_addr))
+        .await
+        .expect("failed to connect to server");
```

Comment:
```suggestion
        .unwrap_or_else(|e| panic!("failed to connect to server {ingester_addr}: {e}"));
```

I think we need to support some ingesters being down when a router starts up (otherwise availability is tied to a single ingester being offline) - I guess this is why there was a deferred connect thing? Perhaps we can error only if all are unreachable or something.

Either way, MVP! MVP! - can you create a ticket for it?

Reply:
Done: https://github.com/influxdata/influxdb_iox/issues/6321